### ISSUE TITLE
GetTransaction with true instead of txindex=1 and fix pruning

### ISSUE THREAD

**Issue by sidhujag** [dashpay/dash#1734]
@UdjinM6  It looks like txindex=1 by default in the later version of Dash to compensate for GetTransaction failing due to pruned blocks. Would it make sense to do txindex=0 and then fix pruning such that it won't prune blocks that have transactions that are part of a masternode collateral? Pruning is non-interactive so no real issues with adding that complexity to the pruning. Does that make sense? I really don't like forcing network wide txindex and removing pruning.

Perhaps another way is to use addressindex=1 and replace the use of GetTransaction although there is a call in instantx for GetTransaction that will not really be replaceable in context and not sure if you have ideas there.

Also see https://github.com/bitcoin/bitcoin/pull/9806 for a txoutindex which is prune compatible to give utxo's while txindex=0 and pruning is enabled. I'm not sure if our version of addressindex is pruning compatible.

**Comment by nmarley** [comment]
I believe this is a duplicate of #1466 and should be fixed once #2014 is merged in.

**Comment by nmarley** [comment]
Closing as this is duplicate, please comment if this is not actually duplicate or should be re-opened for some reason.
